mixer/pkg/api/grpcServer.go

```diff
@@ -161,17 +146,12 @@ func (s *grpcServer) check(legacyCtx legacyContext.Context, req *mixerpb.CheckRe
 
 			crqr := mixerpb.CheckResponse_QuotaResult{}
 
-			var qr *adapter.QuotaResult
+			var qr adapter.QuotaResult
 			qr, err = s.dispatcher.Quota(legacyCtx, checkBag, qma)
 			if err != nil {
 				err = fmt.Errorf("performing quota alloc failed: %v", err)
 				lg.Errora("Quota failure:", err.Error())
 				// we continue the quota loop even after this error
-			} else if qr == nil {
-				// If qma.Quota does not apply to this request give the client what it asked for.
-				// Effectively the quota is unlimited.
-				crqr.ValidDuration = defaultValidDuration
-				crqr.GrantedAmount = qma.Amount
```

Comment:
Can you please explain where this line is now? The default quota should be what was requested, not 0.

Reply:
Well, this question triggered a bunch fo investigation. We have a bug in how we're handling quotas in Mixer at the moment. I'll have to work on that before submitting this change...